File: C++/M01_ProgrammingBasics/L03_ConditionalStatementsAdvanced/Lab/Solutions/P12_TradeCommissions.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {

	string town;
	double sales;
	cin >> town >> sales;

	double commission;
	if (town == "Sofia") {
		if (0 <= sales && sales <= 500) {
			commission = 5;
		} else if (500 < sales && sales <= 1000) {
			commission = 7;
		} else if (1000 < sales && sales <= 10000) {
			commission = 8;
		} else if (10000 < sales) {
			commission = 12;
		} else {
			cout << "error" << endl;
			return 0;
		} 
	} else if (town == "Varna") {
		if (0 <= sales && sales <= 500) {
			commission = 4.5;
		} else if (500 < sales && sales <= 1000) {
			commission = 7.5;
		} else if (1000 < sales && sales <= 10000) {
			commission = 10;
		} else if (10000 < sales) {
			commission = 13;
		} else {
			cout << "error" << endl;
			return 0;
		}
	} else if (town == "Plovdiv") {
		if (0 <= sales && sales <= 500) {
			commission = 5.5;
		}
		else if (500 < sales && sales <= 1000) {
			commission = 8;
		}
		else if (1000 < sales && sales <= 10000) {
			commission = 12;
		}
		else if (10000 < sales) {
			commission = 14.5;
		}
		else {
			cout << "error" << endl;
			return 0;
		}
	} else {
		cout << "error" << endl;
		return 0;
	}

	cout.setf(ios::fixed);
	cout.precision(2);

	cout << (commission / 100) * sales << endl;

	return 0;
}

File: Python/M01_ProgrammingBasics/L03_ConditionalStatementsAdvanced/Lab/Solutions/P12_TradeCommissions.py
city = input()
sales = float(input())

commissions = 0

if city == 'Sofia':
    if 0 <= sales <= 500:
        commissions = 0.05
    elif 500 < sales <= 1000:
        commissions = 0.07
    elif 1000 < sales <= 10000:
        commissions = 0.08
    elif sales > 10000:
        commissions = 0.12
    else:
        print('error')
elif city == 'Varna':
    if 0 <= sales <= 500:
        commissions = 0.045
    elif 500 < sales <= 1000:
        commissions = 0.075
    elif 1000 < sales <= 10000:
        commissions = 0.10
    elif sales > 10000:
        commissions = 0.13
    else:
        print('error')
elif city == 'Plovdiv':
    if 0 <= sales <= 500:
        commissions = 0.055
    elif 500 < sales <= 1000:
        commissions = 0.08
    elif 1000 < sales <= 10000:
        commissions = 0.12
    elif sales > 10000:
        commissions = 0.145
    else:
        print('error')
else:
    print('error')

if city == 'Sofia' or city == 'Varna' or city == 'Plovdiv':
    if sales > 0:
        money = commissions * sales
        print(f'{money:.2f}')
